barretenberg/cpp/src/barretenberg/stdlib_circuit_builders/circuit_simulator.hpp

```diff
@@ -38,6 +40,7 @@ namespace bb {
 class CircuitSimulatorBN254 {
   public:
     using FF = bb::fr;
+    using EmbeddedCurve = std::conditional_t<std::same_as<FF, bb::g1::coordinate_field>, curve::BN254, curve::Grumpkin>;
```

Comment:
where does this come in?

Reply:
we were not explicitly instantiating `cycle_scalar` using the CircuitSimulatorBN254 class. In order to simulate the new stdlib::transcript class (which now has `cycle_scalar` as a dependency) I needed to explicitly instantiate it.

`cycle_scalar` expects the Builder template parameter to have an `EmbeddedCurve` type. This mirrors the code in the other circuit builders.